knora/dsplib/utils/onto_create_lists.py

```diff
@@ -1,43 +1,52 @@
 import json
-from typing import Any, Dict, List
+from typing import Any, Dict, List, Optional
 
 from .expand_all_lists import expand_lists_from_excel
 from .onto_validate import validate_ontology
 from ..models.connection import Connection
+from ..models.helpers import BaseError
 from ..models.listnode import ListNode
 from ..models.project import Project
 
 
-def list_creator(con: Connection, project: Project, parent_node: ListNode, nodes: List[Dict[Any, Any]]) -> List[Dict[Any, Any]]:
+def create_list_node(con: Connection, project: Project, parent_node: ListNode, nodes: List[Dict[Any, Any]]) -> List[
+    Dict[Any, Any]]:
     """
     Creates the list on the DSP server
 
     Args:
-        con: The connection to the DSP server
-        project: The project which the lists should be added
-        parent_node: The root node of the list
+        con: Connection to the DSP server
+        project: Project which the lists should be added to
+        parent_node: Root node of the list
         nodes: List of nodes the list is made of
 
     Returns:
         The list of all nodes with their names and respective IDs
     """
     nodelist = []
     for node in nodes:
-        new_node = ListNode(con=con, project=project, label=node["labels"], comments=node.get("comments"),
-                            name=node["name"],
-                            parent=parent_node).create()
-        if node.get('nodes') is not None:
-            subnode_list = list_creator(con, project, new_node, node['nodes'])
-            nodelist.append({new_node.name: {"id": new_node.id, 'nodes': subnode_list}})
-        else:
-            nodelist.append({new_node.name: {"id": new_node.id}})
+        new_node = None
```

Comment:
It is, because on line 36 it is referenced outside of the try clause which starts on line 29. If I don't declare `new_node` here, I get the warning "local variable 'new_node' might be referenced before assignment".